service/controller/resource/resource.go

```diff
@@ -123,6 +125,31 @@ func New(config Config) ([]resource.Interface, error) {
 		}
 	}
 
+	var remoteWriteIngressResource resource.Interface
+	if config.IngressAPIVersion == "networking.k8s.io/v1beta1" {
```

Comment:
and maybe you can wrap this check in the generic resource file.
If you want to use a generic file